dev/com.ibm.ws.config/src/com/ibm/ws/config/admin/internal/ConfigurationStorageHelper.java

```diff
@@ -0,0 +1,673 @@
+                for (int i = 0; i < numConfigs; i++) {
+                    Set<String> uniqueVars = new HashSet<>();
+                    Set<ConfigID> references = new HashSet<>();
+                    ConfigurationDictionary dict = new ConfigurationDictionary();
+                    String location = load(dis, uniqueVars, references, dict);
+                    result.add(consumer.consumeConfigData(location, uniqueVars, references, dict));
+                }
+                return result;
+            }
+        }
+        return Collections.emptyList();
+    }
+
+    public static String load(DataInputStream dis, Set<String> uniqueVars, Set<ConfigID> references, ConfigurationDictionary dict) throws IOException {
+        String location = readPossiblyNullString(dis);
+        int len = dis.readInt();
+        for (int i = 0; i < len; i++) {
+            uniqueVars.add(dis.readUTF());
+        }
+        len = dis.readInt();
+        List<ConfigID> ids = new ArrayList<>();
+        for (int i = 0, j = 0, cursor = 0; i < len + cursor; i++, j++) {
+            if (cursor == 0) {
+                cursor = dis.readInt();
+                j = 0;
+            }
+            String id = readPossiblyNullString(dis);
+            String configPid = dis.readUTF();
+            String child = readPossiblyNullString(dis);
+            int parentId = dis.readInt();
+            ConfigID parent = null;
+            if (parentId > -1) {
+                parent = ids.get(parentId);
+            }
+            ConfigID thisId = new ConfigID(parent, configPid, id, child);
+            ids.add(thisId);
+
+            if (j == cursor) {
+                len = len + cursor;
+                cursor = 0;
+                references.add(thisId);
+            }
+        }
+        readMap(dis, toMapOrDictionary(dict));
+        return location;
+
+    }
+
+    private static String readPossiblyNullString(DataInputStream dis) throws IOException {
+        if (dis.readBoolean()) {
+            return dis.readUTF();
+        }
+        return null;
+    }
+
+    public static void store(File configFile, Collection<? extends ExtendedConfiguration> configs) throws IOException {
+        try (DataOutputStream dos = new DataOutputStream(new BufferedOutputStream(new FileOutputStream(configFile, false)))) {
+            dos.writeByte(VERSION);
+            dos.writeInt(configs.size());
+            for (ExtendedConfiguration config : configs) {
+                Dictionary<String, ?> properties = config.getReadOnlyProperties();
+                Set<ConfigID> references = config.getReferences();
+                Set<String> uniqueVars = config.getUniqueVariables();
+                String bundleLocation = config.getBundleLocation();
+                store(dos, properties, bundleLocation, references, uniqueVars);
+            }
+        }
+    }
+
+    public static void store(DataOutputStream dos, Dictionary<String, ?> properties, String bundleLocation,
```

Comment:
does this need to be public?

Reply:
No, I made the method package private to allow unit testing.